go/state/state_db.go

```diff
@@ -823,6 +829,33 @@ func (s *stateDB) SetState(addr common.Address, key common.Key, value common.Val
 	}
 }
 
+func (s *stateDB) GetTransientState(addr common.Address, key common.Key) common.Value {
+	sid := slotId{addr, key}
+	val, _ := s.transientStorage.Get(sid)
+	return val
+}
+
+func (s *stateDB) SetTransientState(addr common.Address, key common.Key, value common.Value) {
+	sid := slotId{addr, key}
+	if currentValue, exists := s.transientStorage.Get(sid); exists {
```

Comment:
since non-existing value is treated as empty value, this code could be simplified just adding previous value in the undo lis.  For a non existing value it would be equal to an empty value, and if rollback happens, it would set the empty value in the map.

Reply:
Im not sure if I understood your comment correctly. Please check latest commit.

Reply:
the method can be even simpler: 

```
	sid := slotId{addr, key}
	currentValue, _ := s.transientStorage.Get(sid)
	if currentValue == value {
		return
	}

	// Save previous value for rollbacks
	oldValue := currentValue
	s.undo = append(s.undo, func() {
		s.transientStorage.Put(sid, oldValue)
	})

	s.transientStorage.Put(sid, value)

```